### DISCUSSION TITLE
org.eclipse.gef3 0.0.0 not be found

### DISCUSSION THREAD

**Discussion by xcw233** [dbeaver/dbeaver#22918] [DBeaver Development] [answered]
### Description

I want to build version 22.0.5 from source.
I exchange the website https://dbeaver.io/eclipse-repo to https://p2.dev.dbeaver.com/eclipse-repo/ and I run the building command
`mvn package`
and i failed with 
```
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: org.jkiss.dbeaver.ext.exasol.ui 1.0.51.qualifier
[ERROR]   Missing requirement: org.jkiss.dbeaver.ext.ui.locks 1.0.127.qualifier requires 'osgi.bundle; org.eclipse.gef3 0.0.0' but it could not be found
[ERROR]   Cannot satisfy dependency: org.jkiss.dbeaver.ext.exasol.ui 1.0.51.qualifier depends on: osgi.bundle; org.jkiss.dbeaver.ext.ui.locks 0.0.0
[ERROR]
[ERROR] See https://wiki.eclipse.org/Tycho/Dependency_Resolution_Troubleshooting for help.
[ERROR] Cannot resolve dependencies of MavenProject: org.jkiss.dbeaver:org.jkiss.dbeaver.ext.exasol.ui:1.0.51-SNAPSHOT @ F:\zhibei\Dbeaver\Project\dbeaver\plugins\org.jkiss.dbeaver.ext.exasol.ui\pom.xml: See log for details -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException
```
how can i fix it.
thanks!

### DBeaver Version

22.0.5

### Operating System

windows x86_64

**Comment by Destrolaric** [comment, marked as answer]
@xcw233 Hi, we no longer use this library in recent versions, to solve the issue you can manually build it from https://github.com/dbeaver/gef3

**Comment by xcw233** [comment]
> https://github.com/dbeaver/gef3

@Destrolaric 
Sorry,i don't know how to use it.
How can i get the org.eclipse.gef3 0.0.0 library? 
I have try with https://github.com/dbeaver/gef3 readme to create a eclipse general project and follow it to the end,but i got failed when clicking Set as Target Platform.
```
Problems occurred while resolving the target contents
	Problems loading repositories
		Problems loading repositories
			Unable to locate installable unit org.eclipse.draw2dl.sdk.feature.group 0.0.0
			Unable to locate installable unit org.eclipse.gef3.examples.feature.group 0.0.0
			Unable to locate installable unit org.eclipse.gef3.sdk.feature.group 0.0.0
```
should i do that?
thanks!

**Comment by xcw233** [comment]
> https://p2.dev.dbeaver.com/eclipse-repo/

I have got the gef3 library by package the github gef3 project in org.eclipse.gef3.releng module.
Am i in the right way?
In moudle of org.eclipse.gef3 I can see that.
![image](https://github.com/dbeaver/dbeaver/assets/56435763/3bcbf73e-6cfd-41b7-b0b5-d67219d5aa8c)

**Comment by Destrolaric** [comment]
@xcw233 Hi again, look at dependencies, you may need to compile in Manifest draw2dl and some other libs together with gef3

**Comment by xcw233** [comment]
@Destrolaric  thanks again, i have solved this problem.

**Reply by JoshopSmith29** [reply]
hello, i don't know how to build it, i have set the target platform, what shoud i do next?

**Reply by xcw233** [reply]
Hi,  If you want to build dbeaver, I sugeest you to build the latest version. I have built the 22.0.5 , but that can't run.
https://github.com/orgs/dbeaver/discussions/23000

**Reply by JoshopSmith29** [reply]
thanks, i‘ll try another way :)
